xml_converter/generators/cpp_templates/class_template.hpp

```diff
@@ -36,11 +38,13 @@ class {{cpp_class}}: public Parseable {
         Icon default_icon;
         Trail default_trail;
 
+        virtual vector<string> as_xml() const;
         void init_from_xml(rapidxml::xml_node<>* node, vector<XMLError*> *errors);
+        {%- else %}
+        virtual vector<string> as_xml() const;
```

Comment:
Not sure why there is explicitly duplicated logic now
`if cpp_class == "Category"` you include `virtual vector<string> as_xml() const`
`else` you still include `virtual vector<string> as_xml() const;`
meaning that in all cases it is included.

This change should be undone.

Reply:
Eventually as_xml() will be unique for category because the number of tabs needs to be passed in and out.

Since I reverted my experiments on that function, I will revert it here.

Reply:
While each `as_xml()` function logic will be unique, the function signatures should all be identical.